### PR TITLE
[JENKINS-70928] Remove `Ajax.Request` from breadcrumbs

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-70928](https://issues.jenkins.io/browse/JENKINS-70928) and https://groups.google.com/g/jenkinsci-dev/c/DR9Rr08fd0Y/m/xDWOixaXAwAJ

### Testing done

Clicked around different types of breadcrumbs including nested menus.
I tested cancelling the request by spam clicking and saw browser console errors till I added a catch. (unsure if we really need the cancel but there was something similar before

### Proposed changelog entries

- Entry 1: (too minor?)
- […]

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7727"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by basil — DISMISSED** [review]

**Review by janfaracik — APPROVED** [review]

File: core/src/main/resources/lib/layout/breadcrumbs.js
Comment (basil) [inline core/src/main/resources/lib/layout/breadcrumbs.js:170]: HtmlUnit appears to lack native support for fetch: https://github.com/jenkinsci/jenkins/pull/7731#pullrequestreview-1368116484

Per [the documentation](https://htmlunit.sourceforge.io/webclient.html#Polyfills), "starting with version 2.59.0 HtmlUnit supports the integration of polyfills":

```java
webClient.getOptions().setFetchPolyfillEnabled(true);
```

If core is to begin relying on the Fetch API, then the `WebClient` returned by `createWebClient()` in the Jenkins Test Harness (JTH) needs to have this option set to `true`. Otherwise, every consumer of core (which includes every plugin) will need to repeat the setting of this option to `true` when testing a page that invokes the Fetch API.

**Comment by basil** [comment]
Since this PR has the potential to introduce new test failures by consuming the Fetch API, which is not natively supported by HtmlUnit and which is disabled by default unless a polyfill is enabled (and the Jenkins Test Harness does not currently enable this polyfill), I am requesting a full `jenkinsci/bom` test before this PR is merged.

File: core/src/main/resources/lib/layout/breadcrumbs.js
Comment (timja) [reply core/src/main/resources/lib/layout/breadcrumbs.js:174]: https://github.com/jenkinsci/jenkins-test-harness/pull/567

**Review by basil — APPROVED** [review]
Nice, LGTM from a code perspective. Once BOM tests are verified passing I'll mark this as `ready-for-merge`.

File: core/src/main/resources/lib/layout/breadcrumbs.js
Comment (basil) [inline core/src/main/resources/lib/layout/breadcrumbs.js:93]: Is this really equivalent to the old code? My understanding of the old code is that it would fire off the request and then execute the `if` block without waiting for the response. Your new version waits for the response before executing the `if` block. Was that an intentional change in behavior because you think the old behavior was wrong? My general disposition is that we should minimize changes in behavior to minimize the risk of regression, but I could be persuaded otherwise.

Comment (timja) [reply core/src/main/resources/lib/layout/breadcrumbs.js]: I think an improvement but I've reverted it to minimise change.

**Comment by timja** [comment]
> Nice, LGTM from a code perspective. Once BOM tests are verified passing I'll mark this as `ready-for-merge`.

incrementals didn't get produced 😢.

**Comment by basil** [comment]
Looks like the latest build is green. Did you manage to hit the second `postRequest` function in the debugger in addition to the original testing?

**Comment by timja** [comment]
> Looks like the latest build is green. Did you manage to hit the second `postRequest` function in the debugger in addition to the original testing?

No, the only place I could find that sets it to true was this:
https://github.com/jenkinsci/jenkins/blob/de46d1eebdadf5fbcf1dbedaec8c3471398419c3/core/src/main/java/jenkins/management/ReloadLink.java#L65-L67

but that goes into this if statement and not the else block:

**Comment by timja** [comment]
Bom test is green

**Comment by basil** [comment]
Makes sense, I am sure there is a lot of dead code around these dusty corners.

---

This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
